tests/testthat/test-metrics-sample.R

```diff
@@ -96,7 +96,15 @@ test_that("crps_sample() components correspond to WIS components", {
   expect_true(all(abs(c(dcrps - dwis, ocrps - owis, ucrps - uwis)) < 0.01))
 })
 
-
+test_that("crps_sample() works with a single observation", {
+  expect_no_condition(
+    crps <- crps_sample(
+      observed = 2.5, predicted = 1.5:10.5, separate_results = TRUE
+    )
+  )
+  expect_equal(length(crps), 4)
+  expect_equal(unique(vapply(crps, length, integer(1))), 1)
+})
```

Comment:
```suggestion
})

test_that("crps_sample() works if all values are above or below the median", {
  samples <- 10
  value <- 5
  example <- data.frame(
    observed = rep(value, 2),
    predicted = rnorm(samples * 2, value, 1),
    sample_id = rep(seq_len(samples),  2),
    time =  rep(c("2019-01-01", "2019-01-02"), each = samples)
  )

  expect_no_condition(
    score(as_forecast_sample(example))
  )
})
```
Suggest adding a test for all values below the median

Edit: while writing I realise this is not actually what this is testing, is it? Was that what the original issue was about at all?

Reply:
it's ulimately the same issue - if we have one observation then by definition all observations are either below or above the median. Happy to use the alternative example (tough we'd have to set a seed to ensure they actually all end up above or below the median) but it's more complicated and tests the same thing.

Reply:
Aah didn't realise that. Happy to go with what's there then!